lib/fs.js

```diff
@@ -903,17 +903,27 @@ fs.mkdirSync = function(path, mode) {
                        modeNum(mode, 0o777));
 };
 
-fs.readdir = function(path, callback) {
+fs.readdir = function(path, options, callback) {
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  if (typeof options !== 'object') {
```

Comment:
I'd write it as `options = options || {}`, an explicit typeof check will do the wrong thing when `options === null`.